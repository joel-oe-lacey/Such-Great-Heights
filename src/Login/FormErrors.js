import React from 'react';

export const FormErrors = ({ formErrors }) => {
    return <section className='form-errors'>
                {Object.keys(formErrors).map((error, i) => {
                    if (formErrors[error].length) {
                        return (
                            <p key={i}>{formErrors[error]}</p>
                        )
                    } else {
                        return '';
                    }
                })}
            </section>
}