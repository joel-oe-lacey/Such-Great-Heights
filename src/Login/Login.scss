.login-container {
    font-family: 'Work Sans', sans-serif;
    height: 100%;
    width: 100%;
    display: flex;
    justify-content: center;
    background-color: #89B0AE;
}

.login-form {
  border-radius: 3%;
  padding: 3%;
  height: 80%;
  width: 80%;
  display: grid;
  grid-template-columns: 33.33% 33.33% 33.33%;
  grid-template-rows: auto;
  grid-template-areas:
  '. welcome-login .'
  '. name .'
  '. email .'
  '. type-label .'
  '. travel .'
  '. errors .'
  '. login-button .';
}

.login-welcome {
  grid-area: welcome-login;
  font-size: 2em;
  border-bottom: 2px solid #FFD6BA;
  align-self: flex-end;
}

.name-input {
  grid-area: name;
}

.email-input {
  grid-area: email;
}

.type-label {
  grid-area: type-label;
  align-self: flex-end;
  font-weight: 700;
}

.travel-input {
  grid-area: travel;
  align-self: flex-start;
  font-family: 'Work Sans', sans-serif;
  font-size: 1em;
  height: 40%;
  align-self: center;
  cursor: pointer;
}

.log-in-btn {
  grid-area: login-button;
  border-radius: 4px;
  font-family: 'Work Sans', sans-serif;
  font-weight: 700;
  font-size: 1em;
  background-color: #FFD6BA;
  cursor: pointer;
  padding-right: 20%;
  padding-left: 20%;
  padding-top: 5%;
  padding-bottom: 5%;
  text-decoration: none;
  text-align: center;
  align-self: center;
  color: black;
}

.form-errors {
  grid-area: errors;
  display: flex;
  flex-direction: column;
  align-items: center;
  
  p {
    color: #FFD6BA;
  }
}

input {
  transition: box-shadow 300ms;
  height: 50%;
  font-family: 'Work Sans', sans-serif;
  font-size: 1em;
  height: 40%;
  align-self: center;

  &:focus {
    outline: none;
    border: none;
    box-shadow: 0.1rem 0.3rem 0.6rem black;
  }
}