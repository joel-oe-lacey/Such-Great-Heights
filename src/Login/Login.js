import React, { Component } from 'react';
import './Login.scss';
import { Link } from 'react-router-dom';
import { FormErrors } from './FormErrors';

export default class Login extends Component {
    constructor() {
        super();
        this.state = {
            name:'',
            email:'',
            formErrors: { name: 'Please enter a name', email: 'Please enter an email'},
            nameValid: false,
            emailValid: false,
            formValid: false,
            errorsDisp: false
        }
    }

    submitForm = e => {
        const { updateUser } = this.props;
        updateUser(this.state.name);
    }

    validateField = e => {
        let inputName = e.target.name;
        let value = e.target.value
        let fieldPresentErr = this.state.formErrors;
        let emailValid = this.state.emailValid;
        let nameValid = this.state.nameValid;

        switch (inputName) {
            case 'name':
                nameValid = value;
                fieldPresentErr.name = nameValid ? '' : 'Please enter a name';
                break;
            default:
                emailValid = value;
                fieldPresentErr.email = emailValid ? '' : 'Please enter an email';
                break;
        }

        //call validate form from within this
        //could also do this inline with an || operator to check state or current value  
        this.setState({ 
                [inputName]: value,
                formErrors: fieldPresentErr,
                emailValid,
                nameValid
        }, this.validateForm)
    }

    validateForm = () => {
        this.setState({ formValid: this.state.emailValid && this.state.nameValid });
    }

    showErrors = e => {
        e.preventDefault();
        this.setState({ errorsDisp: true })
    }

    //need a function on button click that does validation based on the state

    //if formValid is false on click, render error messages
    //
    render() {
        const submitBtn = this.state.formValid ? <Link to='/areas' className='log-in-btn'><button className='log-in-btn' onClick={this.submitForm}>Submit</button></Link> : <button className='log-in-btn' onClick={this.showErrors}>Submit</button>;

        const errors = this.state.errorsDisp ? <FormErrors formErrors={this.state.formErrors} /> : '';

        return (
            <section className="login-container">
                <form className="login-form">
                    <h1 className='login-welcome'>Login</h1>
                    <input
                        className='name-input'
                        type="text"
                        name="name"
                        placeholder="Name..."
                        value={this.state.name}
                        onChange={this.validateField}
                    />
                    <input
                        className='email-input'
                        type="text"
                        name="email"
                        placeholder="Email..."
                        value={this.state.email}
                        onChange={this.validateField}
                    />
                    <label className='type-label'>Reason for Traveling</label>
                    <select
                        className='travel-input'
                        name="travelingFor"
                        autoFocus={this.state.travelingFor}
                        onChange={this.validateField}
                    >
                        <option value="business">Business</option>
                        <option value="vacation">Vacation</option>
                        <option value="other">Other</option>
                    </select>
                    {errors}
                    {submitBtn}
                </form>
            </section>
        )
    }
}
