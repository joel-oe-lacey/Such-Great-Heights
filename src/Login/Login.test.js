import React from 'react';
import { shallow } from 'enzyme';
import Login from './Login/Login';

describe('Login', () => {
    it('should equal snapshot ', () => {
        const wrapper = shallow(<Login />);
        expect(wrapper).toMatchSnapshot();
    });

    //need to test
    //use instance and mocks 
    //take listing from details 
        //create a separate mock data file?
    //componentDidMount need to mock fetch

    //selectListing
    //updateUser
    //addFavorite

})