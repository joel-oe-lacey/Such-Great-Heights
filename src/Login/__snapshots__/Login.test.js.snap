// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should equal snapshot  1`] = `
<section
  className="login-container"
>
  <form
    className="login-form"
  >
    <h1
      className="login-welcome"
    >
      Login
    </h1>
    <input
      className="name-input"
      name="name"
      onChange={[Function]}
      placeholder="Name..."
      type="text"
      value=""
    />
    <input
      className="email-input"
      name="email"
      onChange={[Function]}
      placeholder="Email..."
      type="text"
      value=""
    />
    <label
      className="type-label"
    >
      Reason for Traveling
    </label>
    <select
      className="travel-input"
      name="travelingFor"
      onChange={[Function]}
    >
      <option
        value="business"
      >
        Business
      </option>
      <option
        value="vacation"
      >
        Vacation
      </option>
      <option
        value="other"
      >
        Other
      </option>
    </select>
    <button
      className="log-in-btn"
      onClick={[Function]}
    >
      Submit
    </button>
  </form>
</section>
`;
